Standards/scs-XXXX-v1-encrypted-internal-openstack-traffic-and-pki-infra.md

```diff
@@ -0,0 +1,260 @@
+13. **OpenvSwitch** - OpenvSwitch is a multilayer virtual switch implemented to ensure effective network automation in an OpenStack environment.
+
+
+### Current state
+
+#### Non-internal OpenStack endpoints
+
+##### OpenStack API: External client
+
+```
+   EC            --|TLS|--         HA          --|HTTP|--               OS
+(External)   (public internet) (HAProxy) (OpenStack Networking)   (OpenStack Services)
+   |                                                                  |      |      |
+ Client                                                           Keystone  Glance  Placement ...
+```
+
+The communication between external clients and the OpenStack services is crucial. It needs to be secured to prevent data leakage and unauthorized access as this data often travels across the open internet, is highly susceptible to interception, and has considerable potential for misuse.
+
+Kolla, out of the box, provides the ability to encrypt external traffic using a HAProxy. [Details](https://docs.openstack.org/kolla-ansible/latest/admin/tls.html#tls-configuration-for-internal-external-vip)
+
+##### OpenStack API: Internal client
+
+```
+   IC            --|TLS|--         HA          --|HTTP|--               OS
+(Internal) (OpenStack Networking) (HAProxy) (OpenStack Networking)   (OpenStack Services)
+   |                                                                  |      |      |
+ Client                                                          Keystone  Glance  Placement ...
+```
+
+The communication from within the OpenStack is the second most vulnerable communication. This kind of communication usually serves to manage resources from CI/CD tools or services such as the SCS KaaS layer. For the attacker to gain access, they can intercept this kind of communication, for example, from a compromised ci build.
+
+Kolla, out of the box, can encrypt internal traffic using a HAProxy. [Details](https://docs.openstack.org/kolla-ansible/latest/admin/tls.html#tls-configuration-for-internal-external-vip)
+
+##### HAProxy to backend services
+
+```
+   EC            --|TLS|--         HA          --|TLS|--                   OS
+(External)   (public internet) (HAProxy) (OpenStack Networking)   (OpenStack Services)
+   |                                                                  |      |      |
+ Client                                                           Keystone  Glance  Placement ...
+```
+
+The communication from HAproxy to respective services is much harder to intercept as this traffic is internal to OpenStack; however, here, an attacker is possible from one compromised service to intercept all API traffic. 
+
+Kolla enables end-to-end TLS encryption from HAProxy to services that support TLS Termination, currently out of the services of interest, the following services are supported:
+
+1. Keystone
+2. Horizon
+3. Glance
+4. Cinder
+5. Placement
+6. Nova
+7. Netron
+8. Heat
+13. OpenVSwitch
+
+Only RabbitMQ is not currently supported.
```

Comment:
there is interest in adding an internal CA for rabbitmq and libvirt TLS support, see:

https://etherpad.opendev.org/p/kolla-caracal-ptg#L283

the problematic part currently is, choosing the appropriate CA software, which has many challenges, as can be seen in the linked document where we evaluated various alternatives.